File: leetcode/404.swift

class Solution {
    func sumOfLeftLeaves(_ root: TreeNode?) -> Int {
        if root == nil { return 0 }
        
        var result = 0
        var nodes: [TreeNode] = [root!]
        
        while nodes.count > 0 {
            let node = nodes[0]
            
            if node.left != nil &&
                node.left?.left == nil && node.left?.right == nil {
                result += node.left?.val ?? 0
            }
            
            nodes.remove(at: 0)
            
            if node.left != nil { nodes.append(node.left!) }
            if node.right != nil { nodes.append(node.right!) }
        }
        
        return result
    }
}

File: leetcode/404.java

class Solution {

    int result = 0;

    void find(TreeNode node) {
        if(node.left != null &&
                node.left.left == null && node.left.right == null) {
            result += node.left.val;
        }

        if(node.left != null) { find(node.left); }
        if(node.right != null) { find(node.right); }
    }

    public int sumOfLeftLeaves(TreeNode root) {
        if(root == null) { return 0; }

        find(root);
        return result;
    }
}